### PR TITLE
[Feature/346567: Overseas reprocessor details]

### PR DESCRIPTION
**Pull request by shehzad-ismail8**

### CODE REVIEW

File: src/EPR.Accreditation.Facade.Common/RESTservices/HttpCountryService.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            IHttpContextAccessor httpContextAccessor,
+            IHttpClientFactory httpClientFactory,
+            string baseUrl,
+            string endPointName) : base(httpContextAccessor, httpClientFactory, baseUrl, endPointName)
+        {
+        }
+
+        public async Task<IEnumerable<Country>> GetCountryList()
+        {
+            return await Get<IEnumerable<Country>>("", false);
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade.Common/RESTservices/HttpCountryService.cs]: Use string.Empty not double quotes

File: src/EPR.Accreditation.Facade/Controllers/OverseasSiteController.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            await _overseasSiteService.UpdateReprocessorDetails(
+                accreditationExternalId,
+                overseasSiteExternalId,
+                reprocessorDetails);
+
+            return Ok();
+        }
+    }
+
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Controllers/OverseasSiteController.cs]: Blank line

File: src/EPR.Accreditation.Facade/Services/CountryService.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        public CountryService(IHttpCountryService httpCountryService)
+        {
+            _httpCountryService = httpCountryService ?? throw new ArgumentNullException(nameof(httpCountryService));
+        }
+
+        public async Task<IEnumerable<Country>> GetCountryList()
+        {
+            var listOfCountries = await _httpCountryService.GetCountryList();
+
+            if (listOfCountries == null)
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Services/CountryService.cs]: This isn't necessary

File: src/EPR.Accreditation.Facade/Services/OverseasSiteService.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        public async Task<OverseasAddress> GetReprocessorDetails(
+            Guid accreditationExternalId,
+            Guid overseasSiteExternalId)
+        {
+            var overseasSite = await _httpOverseasSiteService.GetOverseasReprocessingSite(
+                accreditationExternalId,
+                overseasSiteExternalId);
+
+            if (overseasSite.OverseasAddress == null)
+                return null;
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Services/OverseasSiteService.cs:26]: StyleCop for Portal requires braces even for one line if stamenents. Let's continue that style

File: src/EPR.Accreditation.Facade/Services/OverseasSiteService.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            var overseasSite = new OverseasReprocessingSite
+            {
+                ExternalId = overseasSiteExternalId,
+                OverseasAddress = reprocessorDetails
+            };
+
+            await _httpOverseasSiteService.UpdateOverseasReprocessingSite(
+                accreditationExternalId,
+                overseasSite
+                );
```
Comment (NikPeters-Atos) [inline src/EPR.Accreditation.Facade/Services/OverseasSiteService.cs]: Don't have a bracket all by itself

File: src/EPR.Accreditation.Facade.Common/RESTservices/HttpCountryService.cs (hunk 1 above)
Comment (shehzad-ismail8) [reply src/EPR.Accreditation.Facade.Common/RESTservices/HttpCountryService.cs]: Changed

File: src/EPR.Accreditation.Facade/Controllers/OverseasSiteController.cs (hunk 2 above)
Comment (shehzad-ismail8) [reply src/EPR.Accreditation.Facade/Controllers/OverseasSiteController.cs]: Removed

File: src/EPR.Accreditation.Facade/Services/CountryService.cs (hunk 3 above)
Comment (shehzad-ismail8) [reply src/EPR.Accreditation.Facade/Services/CountryService.cs]: Removed

File: src/EPR.Accreditation.Facade/Services/OverseasSiteService.cs (hunk 4 above)
Comment (shehzad-ismail8) [reply src/EPR.Accreditation.Facade/Services/OverseasSiteService.cs:26]: Added

File: src/EPR.Accreditation.Facade/Services/OverseasSiteService.cs (hunk 5 above)
Comment (shehzad-ismail8) [reply src/EPR.Accreditation.Facade/Services/OverseasSiteService.cs]: Changed

**Review by NikPeters-Atos — APPROVED** [review]

**Pull request merged**